### ISSUE TITLE
support linux path style with / 

### ISSUE THREAD

**Issue by kbilsted** [dnGrep/dnGrep#1240]
it should be possible to find files in folder `src` both using `C:\src` and `C:/src` this is not the case today. the linux style of specifying a folder is supported many places in windows already, and i think also c# actually has support for it out of the box.

**Comment by doug24** [comment]
Yes, using slashes does work with some libraries like System.IO which have been made cross-platform, but it doesn't work in all cases.  dnGrep uses native Win32 calls to enumerate directories and files because it is so much faster than using similar .net calls - and it doesn't work there.  Or if you are using Everything to find files, it doesn't recognize slashes as path separators.

The only way I would consider doing this is if the code immediately converted all the slashes to backslashes on user input, so all the code internally is only using backslashes.  But even that is a big task affecting a lot of code.

Is this really a problem? Is there a file system you can't search from Windows using backslashes?

**Comment by kbilsted** [comment]
In my setup it is easier to press `/` and since it is working in all other programs I thought it would be nice to do the same here. A simple search/replace when processing the input is perhaps the easier way to go.. if you decide to implement it.

**Comment by kbilsted** [comment]
perhaps it is somewhere else the time is spent than what you conjecture?
![image](https://github.com/user-attachments/assets/876f892f-56bd-4967-9780-ea8cc19439f0)


compare this to what i just threw together

```
using System.Diagnostics;

namespace filefinder
{
    internal class Program
    {
        static IEnumerable<string> files(string path)
        {
            if (path.StartsWith("."))
                yield break;

            foreach (var d in Directory.EnumerateFiles(path, "*.cs", SearchOption.TopDirectoryOnly))
                yield return d;
            foreach (var d in Directory.EnumerateDirectories(path, "*", SearchOption.TopDirectoryOnly))
                foreach (var f in files(d))
                    yield return f;
        }

        static void Main(string[] args)
        {
            var sw = Stopwatch.StartNew();
            var entries = files(@"c:/src")
               .AsParallel()
               .Select(x => new { x, lines = File.ReadAllLines(x).Where(y => y.Contains("complete")).ToArray() });

            int i = 1;
            foreach (var entry in entries)
                foreach (var k in entry.lines)
                    Console.WriteLine(i++ + k);
            Console.WriteLine($"{sw.ElapsedMilliseconds} millis");
        }
    }
}
```

runtime on the same folder: 1130 millis.. around 4 x faster.

**Comment by doug24** [comment]
I ran your program, and I'm not seeing the same results as you show.  Even with the additional complexity dnGrep has to handle, it is faster than your program on my machine. 

![image](https://github.com/user-attachments/assets/499ec0ba-d127-422c-ba51-ced032a73a34)

Your program on .NET 8, Release build:  6728 millis
I don't see why it would run so much faster on your machine - what did you do different?

Some settings you (may) have in dnGrep that will slow it down (but not 4x):
Check Include Hidden (checking for hidden files takes time)
Check Include Binary (checking if a file is binary takes time)
Uncheck Use .gitignore (finding, processing and applying exclusion filters takes time)
Set the Encoding to UTF-8 (running the algorithm to find the encoding takes time)

But even with these set, dnGrep still finishes in 6519ms.

**Comment by kbilsted** [comment]
I will try setting more to off. 
Try installing ripgreb it is x2 faster than yours.

**Comment by doug24** [comment]
Yes, I am very familiar with ripgrep, it's very good. I recommend it when people ask for a [full command line interface](https://github.com/dnGrep/dnGrep/issues/1015)  (no UI) for dnGrep (not something I'll do). I also had the request to make dnGrep a front end for ripgrep. Also, not something I'm interested in doing. 

I'm sure there must still things that can be optimized in dnGrep, but I've spent a lot of time on that already and others have looked at it too. 

Need to go faster?  Use [Everything](https://www.[voidtools](https://www.voidtools.com/).com/) in dnGrep for file indexing - here is the same search in 825 ms - 5x faster.

![image](https://github.com/user-attachments/assets/4b2c1da2-6efa-40ab-a676-f3ff7bda1878)
